components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/handlers/AbstractFieldValidationConfigurationHandler.java

```diff
@@ -68,14 +69,14 @@ protected RulesConfiguration getRuleConfig(String validatorName, String property
     protected RealmConfiguration getRealmConfiguration(String tenantDomain) throws InputValidationMgtException {
 
         try {
-            // Retrieve configs from secondary userstore if configured in the identity.xml, else from PRIMARY userstore.
             UserRealm realm = (UserRealm) CarbonContext.getThreadLocalCarbonContext().getUserRealm();
-            String secondaryUserStoreName =
-                    IdentityUtil.getProperty("FieldInputValidation.DefaultConfigurations.SecondaryUserStoreName");
-            if (realm != null && StringUtils.isNotBlank(secondaryUserStoreName)
-                    && realm.getUserStoreManager().getSecondaryUserStoreManager(secondaryUserStoreName) != null) {
+            // Retrieve configs from secondary userstore if the tenant domain is not carbon.super.
+            // This logic should be improved when exposing the API to product-is.
+            if (realm != null && !tenantDomain.equals(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME) &&
+                    StringUtils.isNotBlank(SECONDARY_USER_STORE_DOMAIN_NAME) && realm.getUserStoreManager().
```

Comment:
Do we need to check whether `SECONDARY_USER_STORE_DOMAIN_NAME` is null or not as it is a constant?

Reply:
As I added a constant (static final) it can not be null and value can not be changed.

Reply:
Fixed